locator/snitch_base.cc

```diff
@@ -37,8 +37,23 @@
  */
 
 #include "locator/snitch_base.hh"
+#include "gms/gossiper.hh"
+#include "gms/application_state.hh"
 
 namespace locator {
+std::optional<sstring>
+snitch_base::get_endpoint_info(inet_address endpoint,
```

Comment:
I guess get_endpoint_info could be just a standalone function. I extracted it from production_snitch_base::get_endpoint_info because I needed exactly the same logic in get_shard_count. I thought that it would be clean to extract a common logic to a base class and then use it both in a base class and subclass. I can make it static if you like. Or turn it into a fully standalone function if you insist. Which one do you prefer?

Reply:
I don't know what I prefer, I think I still don't understand what this function is trying to do. If get_endpoint_info() isn't virtual, and there are two implementations, which one will be called from get_shard_count()? Does making get_shard_count() virtual, but with just a single implementation, help?

Reply:
snitch_base::get_endpoint_info is not virtual (and I made it static) on purpose. It's a helper method that's used in snitch_base::get_shard_count and in production_snitch_base::get_endpoint_info. production_snitch_base::get_endpoint_info and snitch_base::get_endpoint_info are not related. Then only relation is that production_snitch_base::get_endpoint_info uses snitch_base::get_endpoint_info.

Reply:
Ok. I find this very confusing but as long as you checked that this actually does what you want, I'll believe you :-)